src/main/java/DemandeAvenantComponent.java

```diff
@@ -0,0 +1,433 @@
+package fr.aba.prevoyance.api.component;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import fr.aba.prevoyance.api.converter.ContratConverter;
+import fr.aba.prevoyance.api.converter.DemandeAvenantConverter;
+import fr.aba.prevoyance.postgresql.entity.DemandeAvenantContratEntity;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import fr.aba.prevoyance.api.service.DemandeAvenantService;
+import com.aviva.prevoyance.pivot.adhesion.contrat.avenant.DemandeAvenantPivot;
+import com.aviva.prevoyance.pivot.adhesion.contrat.v2.ContratPivot;
+import com.aviva.prevoyance.pivot.couverture.v2.CouverturePivot;
+import com.aviva.prevoyance.pivot.couverture.v2.GarantiePivot;
+import com.aviva.prevoyance.pivot.finance.PrimePivot;
+import com.aviva.prevoyance.pivot.finance.SurprimePivot;
+import com.aviva.prevoyance.pivot.personne.v2.RolePersonnePivot;
+import com.aviva.prevoyance.pivot.v2.DossierPivot;
+import com.aviva.prevoyance.pivotparametrage.IntervalleMontantsPivot;
+import com.aviva.prevoyance.pivotparametrage.avenant.DemandeAvenantConstantes;
+import com.aviva.prevoyance.pivotparametrage.avenant.DemandeAvenantConstantes.EtatDemandeAvenant;
+import com.aviva.prevoyance.pivottechnical.exceptions.v2.ComponentCoherenceDatasException;
+import com.aviva.prevoyance.pivottechnical.exceptions.v2.ComponentValidationException;
+import com.aviva.prevoyance.pivottechnical.exceptions.v2.EnumNotFoundException;
+import fr.aba.prevoyance.postgresql.entity.DemandeAvenantEntity;
+import fr.aba.prevoyance.postgresql.repository.DemandeAvenantContratRepository;
+import fr.aba.prevoyance.postgresql.repository.DemandeAvenantRepository;
+import com.aviva.prevoyance.utils.component.BigDecimalComponent;
+import com.aviva.prevoyance.utilsmicroservice.component.contrat.APIGarantie;
+
+@Component
+public class DemandeAvenantComponent {
+
+	@Autowired private DemandeAvenantRepository repositoryDemandeAvenant;
+	@Autowired private DemandeAvenantContratRepository repositoryDemandeAvenantContrat;
+	@Autowired private DemandeAvenantConverter convDemandeAvenant;
+	@Autowired private DemandeAvenantService sDemandeAvenant;
+	@Autowired private APIGarantie apiGarantie;
+
+	@Autowired private GarantieComponent cGarantie;
+	@Autowired private PersonneComponent cPersonne;
+	@Autowired private SurprimesComponent surprimesComponent;
```

Comment:
Direct field injection is discouraged as it violates the Dependency Inversion Principle (DIP). Consider using constructor injection instead. (Severity: medium)